cmd/subcmd/touch.go

```diff
@@ -0,0 +1,136 @@
+package subcmd
+
+import (
+	irodsclient_fs "github.com/cyverse/go-irodsclient/fs"
+	irodsclient_types "github.com/cyverse/go-irodsclient/irods/types"
+	"github.com/cyverse/gocommands/cmd/flag"
+	"github.com/cyverse/gocommands/commons"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"golang.org/x/xerrors"
+)
+
+var touchCmd = &cobra.Command{
+	Use:     "touch [data-object]",
+	Aliases: []string{"itouch"},
+	Short:   "Create an empty iRODS data-object or update last modified time of existing data-object",
+	Long:    `This displays the content of an iRODS data-object.`,
+	RunE:    processTouchCommand,
+	Args:    cobra.MinimumNArgs(1),
+}
+
+func AddTouchCommand(rootCmd *cobra.Command) {
+	// attach common flags
+	flag.SetCommonFlags(touchCmd, false)
+
+	flag.SetTicketAccessFlags(touchCmd)
+	flag.SetNoCreateFlags(touchCmd)
+
+	rootCmd.AddCommand(touchCmd)
+}
+
+func processTouchCommand(command *cobra.Command, args []string) error {
+	touch, err := NewTouchCommand(command, args)
+	if err != nil {
+		return err
+	}
+
+	return touch.Process()
+}
+
+type TouchCommand struct {
+	command *cobra.Command
+
+	ticketAccessFlagValues *flag.TicketAccessFlagValues
+	noCreateFlagValues     *flag.NoCreateFlagValues
+
+	account    *irodsclient_types.IRODSAccount
+	filesystem *irodsclient_fs.FileSystem
+
+	targetPaths []string
+}
+
+func NewTouchCommand(command *cobra.Command, args []string) (*TouchCommand, error) {
+	touch := &TouchCommand{
+		command: command,
+
+		ticketAccessFlagValues: flag.GetTicketAccessFlagValues(),
```

Comment:
tickets on the touch interface?

Reply:
I thought it will be okay as it's simple data modification API call. But it may not. I removed it.